packages/page-accounts/src/Accounts/modals/Qr.tsx

```diff
@@ -26,6 +28,15 @@ function QrModal ({ className = '', onClose, onStatusChange }: Props): React.Rea
   const { t } = useTranslation();
   const [{ isNameValid, name }, setName] = useState({ isNameValid: false, name: '' });
   const [scanned, setScanned] = useState<Scanned | null>(null);
+  const [isAddress, setIsAddress] = useState<boolean>(false);
+  const [address, setAddress] = useState<string>('');
+  const defaultPasswordState = { isPassValid: false, password: '' };
+  const passwordStateHooks = useState(defaultPasswordState);
+  const [{ isPassValid, password }] = passwordStateHooks;
+  const password2StateHooks = useState(defaultPasswordState);
+  const [{ isPassValid: isPass2Valid }] = password2StateHooks;
+
+  const isValid = isAddress ? (!!address && isNameValid) : (!!address && isNameValid && isPassValid && isPass2Valid);
```

Comment:
Alternative without supes...

```suggestion
  const isValid = !!address && isNameValid && (isAddress || (isPassValid && isPass2Valid));
```